staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -341,10 +331,6 @@ func (d *DiscoveryClient) ServerResourcesForGroupVersion(groupVersion string) (r
 	}
 	err = d.restClient.Get().AbsPath(url.String()).Do(context.TODO()).Into(resources)
 	if err != nil {
-		// ignore 403 or 404 error to be compatible with an v1.0 server.
-		if groupVersion == "v1" && (errors.IsNotFound(err) || errors.IsForbidden(err)) {
-			return resources, nil
-		}
```

Comment:
I think this error makes sense to propagate if you're asking for a specific group/version and the request failed, but we need to sweep callers and ensure returning NotFound or Forbidden errors won't break anything built on this

these stand out:
```go
				apiResourceList, err := d.ServerResourcesForGroupVersion(groupVersion.String())

				// lock to record results
				resultLock.Lock()
				defer resultLock.Unlock()

				if err != nil {
					// TODO: maybe restrict this to NotFound errors
					failedGroups[groupVersion] = err
				}
```

```go
func (dc *DisruptionController) implementsScale(gvr schema.GroupVersionResource) (bool, error) {
	resourceList, err := dc.discoveryClient.ServerResourcesForGroupVersion(gvr.GroupVersion().String())
	if err != nil {
		return false, err
	}

	scaleSubresourceName := fmt.Sprintf("%s/scale", gvr.Resource)
	for _, resource := range resourceList.APIResources {
		if resource.Name != scaleSubresourceName {
			continue
		}

		for _, scaleGv := range scaleclient.NewScaleConverter().ScaleVersions() {
			if resource.Group == scaleGv.Group &&
				resource.Version == scaleGv.Version &&
				resource.Kind == "Scale" {
				return true, nil
			}
		}
	}
	return false, nil
}
```

Reply:
1) There is no change in behavior for the `DisruptionController.implementsScale` function if we propagate the discovery error--`false` is always returned. If we propagate the discovery error, this function returns `false` immediately. If not, the `resourceList` is empty, and we fall-through to `false`.

Reply:
2. Within the discovery client (e.g. `ServerGroupsAndResources` and `ServerPreferredResources`), propagating this "core/v1" error does not change the list of resources returned--but it will change the list of Groups returned (the "core" group will not be in it), and it will propagate an error (with the failedGroups including the core group). Arguably, this is more **correct**.

Reply:
3. Another example of "core/v1" discovery request in `drain.go`, where behavior does not change. If the 
 discovery error is propagated, or the current empty resourceList is returned, the function returns the same empty `schema.GroupVersion`.
```
        resourceList, err := discoveryClient.ServerResourcesForGroupVersion("v1")                                                                                 
        if err != nil {                                                                                                                                           
                return schema.GroupVersion{}, err                                                                                                                 
        }                                                                                                                                                         
        for _, resource := range resourceList.APIResources {                                                                                                      
                if resource.Name == EvictionSubresource && resource.Kind == EvictionKind && len(resource.Group) > 0 && len(resource.Version) > 0 {                
                        return schema.GroupVersion{Group: resource.Group, Version: resource.Version}, nil                                                         
                }                                                                                                                                                 
        }                                                                                                                                                         
        return schema.GroupVersion{}, nil        ```